File: src/platform/ios/LevelWaveRequirementMapper.m
//
//  LevelWaveRequirementMapper.m
//  insectoid-defense
//
//  Created by Stephen Gowen on 7/24/13.
//  Copyright (c) 2014 Gowen Game Dev. All rights reserved.
//

#import "LevelWaveRequirementMapper.h"

@implementation LevelWaveRequirementMapper

+ (int)getWaveRequirementForLevelIndex:(int)levelIndex
{
    switch(levelIndex)
    {
        case 0:
            return 20;
        case 1:
            return 30;
        case 2:
            return 30;
        case 3:
            return 30;
        case 4:
            return 20;
        case 5:
            return 30;
        case 6:
            return 20;
        case 7:
            return 50;
        case 8:
            return 40;
        case 9:
            return 60;
        default:
            return 1337;
    }
}

+ (NSString *)getLockedDescriptionStringForLevelIndex:(int)levelIndex
{
    switch(levelIndex)
    {
        case 1:
            return NSLocalizedString(@"level_2_locked", nil);
        case 2:
            return NSLocalizedString(@"level_3_locked", nil);
        case 3:
            return NSLocalizedString(@"level_4_locked", nil);
        case 4:
            return NSLocalizedString(@"level_5_locked", nil);
        case 5:
            return NSLocalizedString(@"level_6_locked", nil);
        case 6:
            return NSLocalizedString(@"level_7_locked", nil);
        case 7:
            return NSLocalizedString(@"level_8_locked", nil);
        case 8:
            return NSLocalizedString(@"level_9_locked", nil);
        case 9:
            return NSLocalizedString(@"level_10_locked", nil);
        default:
            return @"";
    }
}

+ (NSString *)getLevelUnLockedMessageStringForLevelIndex:(int)levelIndex
{
    switch(levelIndex)
    {
        case 0:
            return NSLocalizedString(@"level_2_unlocked", nil);
        case 1:
            return NSLocalizedString(@"level_3_unlocked", nil);
        case 2:
            return NSLocalizedString(@"level_4_unlocked", nil);
        case 3:
            return NSLocalizedString(@"level_5_unlocked", nil);
        case 4:
            return NSLocalizedString(@"level_6_unlocked", nil);
        case 5:
            return NSLocalizedString(@"level_7_unlocked", nil);
        case 6:
            return NSLocalizedString(@"level_8_unlocked", nil);
        case 7:
            return NSLocalizedString(@"level_9_unlocked", nil);
        case 8:
            return NSLocalizedString(@"level_10_unlocked", nil);
        case 9:
            return NSLocalizedString(@"game_cleared", nil);
        default:
            return @"";
    }
}

@end

File: src/platform/android/app/src/main/java/com/gowengamedev/insectoiddefense/LevelWaveRequirementMapper.java

package com.gowengamedev.insectoiddefense;

import android.annotation.SuppressLint;

import java.util.HashMap;
import java.util.Map;

@SuppressLint("UseSparseArrays")
public final class LevelWaveRequirementMapper
{
    private static final Map<Integer, Integer> levelWaveRequirementMap;
    private static final Map<Integer, Integer> levelLockedDescriptionStringResourceMap;
    private static final Map<Integer, Integer> levelUnlockedAchievementTextMap;

    static
    {
        levelWaveRequirementMap = new HashMap<>();
        levelWaveRequirementMap.put(0, 20);
        levelWaveRequirementMap.put(1, 30);
        levelWaveRequirementMap.put(2, 30);
        levelWaveRequirementMap.put(3, 30);
        levelWaveRequirementMap.put(4, 20);
        levelWaveRequirementMap.put(5, 30);
        levelWaveRequirementMap.put(6, 20);
        levelWaveRequirementMap.put(7, 50);
        levelWaveRequirementMap.put(8, 40);
        levelWaveRequirementMap.put(9, 60);

        levelLockedDescriptionStringResourceMap = new HashMap<>();
        levelLockedDescriptionStringResourceMap.put(1, R.string.level_2_locked);
        levelLockedDescriptionStringResourceMap.put(2, R.string.level_3_locked);
        levelLockedDescriptionStringResourceMap.put(3, R.string.level_4_locked);
        levelLockedDescriptionStringResourceMap.put(4, R.string.level_5_locked);
        levelLockedDescriptionStringResourceMap.put(5, R.string.level_6_locked);
        levelLockedDescriptionStringResourceMap.put(6, R.string.level_7_locked);
        levelLockedDescriptionStringResourceMap.put(7, R.string.level_8_locked);
        levelLockedDescriptionStringResourceMap.put(8, R.string.level_9_locked);
        levelLockedDescriptionStringResourceMap.put(9, R.string.level_10_locked);

        levelUnlockedAchievementTextMap = new HashMap<>();
        levelUnlockedAchievementTextMap.put(0, R.string.level_2_unlocked);
        levelUnlockedAchievementTextMap.put(1, R.string.level_3_unlocked);
        levelUnlockedAchievementTextMap.put(2, R.string.level_4_unlocked);
        levelUnlockedAchievementTextMap.put(3, R.string.level_5_unlocked);
        levelUnlockedAchievementTextMap.put(4, R.string.level_6_unlocked);
        levelUnlockedAchievementTextMap.put(5, R.string.level_7_unlocked);
        levelUnlockedAchievementTextMap.put(6, R.string.level_8_unlocked);
        levelUnlockedAchievementTextMap.put(7, R.string.level_9_unlocked);
        levelUnlockedAchievementTextMap.put(8, R.string.level_10_unlocked);
        levelUnlockedAchievementTextMap.put(9, R.string.game_cleared);
    }

    public static int getWaveRequirementForLevelIndex(Integer levelIndex)
    {
        return levelWaveRequirementMap.get(levelIndex);
    }

    public static int getLockedDescriptionStringResourceForLevelIndex(Integer levelIndex)
    {
        return levelLockedDescriptionStringResourceMap.get(levelIndex);
    }

    public static int getLevelUnlockedAchievementToastStringResourceForLevelIndex(Integer levelIndex)
    {
        return levelUnlockedAchievementTextMap.get(levelIndex);
    }
}